libfwupd/fwupd-common.c

```diff
@@ -133,3 +135,115 @@ fwupd_checksum_get_best (GPtrArray *checksums)
 	}
 	return NULL;
 }
+
+static gchar *
+fwupd_build_user_agent_os_release (void)
+{
+	const gchar *keys[] = { "NAME=", "VERSION_ID=", "VARIANT=", NULL };
+	const gchar *values[] = { NULL, NULL, NULL, NULL };
+	g_autofree gchar *os_release = NULL;
+	g_auto(GStrv) lines = NULL;
+	g_autoptr(GPtrArray) ids_os = g_ptr_array_new ();
+
+	/* get raw data then parse each line */
+	if (!g_file_get_contents ("/etc/os-release", &os_release, NULL, NULL))
```

Comment:
I think you should set up a fall back according to freedesktop.

https://www.freedesktop.org/software/systemd/man/os-release.html

Reply:
Good catch, fixed.